test/index.ts

```diff
@@ -1,805 +1,1080 @@
-          t.equals(blocks.length, 5, 'should get 5 blocks')
-          t.ok(isConsecutive(blocks.reverse()), 'blocks should be consecutive')
-          done()
+        blockchain.getBlocks(5, 5, 0, true, (err?: any, getBlocks?: any) => {
+          st.error(err, 'no error')
+          st.equal(getBlocks.length, 5)
+          st.ok(getBlocks[0].header.number.equals(blocks[5].header.number))
+          st.ok(isConsecutive(getBlocks.reverse()), 'blocks should be consecutive')
+          st.end()
         })
-      },
-      function getBlocks11(done) {
+      })
+    })
+  })
+
+  t.test('should get 6 consecutive blocks, starting from block 5, reversed', st => {
+    const blockchain = new Blockchain({ validate: false })
+    const blocks: any[] = []
+    const genesisBlock = new Block()
+    genesisBlock.setGenesisParams()
+    blocks.push(...generateBlocks(15, genesisBlock))
+    blockchain.putGenesis(genesisBlock, (err?: Error) => {
+      st.error(err, 'no error')
+      blockchain.putBlocks(blocks.slice(1), (err?: Error) => {
+        if (err) {
+          return st.error(err)
+        }
         // start: 5, max: 10, skip: 0, reverse: true
-        blockchain.getBlocks(5, 10, 0, true, function(err?: any, blocks?: any) {
-          if (err) {
-            return done(err)
-          }
-          t.equals(blocks.length, 6, 'should get 6 blocks')
-          t.ok(isConsecutive(blocks.reverse()), 'blocks should be consecutive')
-          done()
+        blockchain.getBlocks(5, 10, 0, true, (err?: any, getBlocks?: any) => {
+          st.error(err, 'no error')
+          st.equal(getBlocks.length, 6)
+          st.ok(getBlocks[0].header.number.equals(blocks[5].header.number))
+          st.ok(isConsecutive(getBlocks.reverse()), 'blocks should be consecutive')
+          st.end()
         })
-      },
-      function getBlocks12(done) {
-        // start: 5, max: 10, skip: 0, reverse: true
-        blockchain.getBlocks(5, 10, 1, true, function(err?: any, blocks?: any) {
-          if (err) {
-            return done(err)
-          }
-          t.equals(blocks.length, 3, 'should get 3 blocks')
-          t.ok(!isConsecutive(blocks.reverse()), 'blocks should not be consecutive')
-          done()
+      })
+    })
+  })
+
+  t.test('should get 3 blocks, skipping 1, starting from block 5, reversed', st => {
+    const blockchain = new Blockchain({ validate: false })
+    const blocks: any[] = []
+    const genesisBlock = new Block()
+    genesisBlock.setGenesisParams()
+    blocks.push(...generateBlocks(15, genesisBlock))
+    blockchain.putGenesis(genesisBlock, (err?: Error) => {
+      st.error(err, 'no error')
+      blockchain.putBlocks(blocks.slice(1), (err?: Error) => {
+        if (err) {
+          return st.error(err)
+        }
+        // start: 5, max: 5, skip: 1, reverse: true
+        blockchain.getBlocks(5, 5, 1, true, (err?: any, getBlocks?: any) => {
+          st.error(err, 'no error')
+          st.equal(getBlocks.length, 3)
```

Comment:
It should equal 5 after #137 is fixed. Or `getBlocks(10, 10, 1, ...)`, in which case the length should be 4 (I think)